app/jobs/notifications/remove_all_by_action_job.rb

```diff
@@ -0,0 +1,9 @@
+module Notifications
+  class RemoveAllByActionJob < ApplicationJob
+    queue_as :remove_all_by_action_notifications
+
+    def perform(notifiable, action, service = Notifications::RemoveAllByAction)
```

Comment:
`notifiable` should be a pair of `notifiable_type` and `notifiable_id`, like in other job classes the job itself should load the object from the DB and do nothing if the object doesn't exist. This avoid deserialization errors and the corner case of the notifiable that has been deleted before this job is called